src/main/java/com/mixpanel/android/mpmetrics/Tweaks.java

```diff
@@ -1,68 +1,71 @@
- *
- * For example, you may want to add an advertisement to your application, but make sure it doesn't hurt
- * your retention before rolling it out to all of your users. Early in your application's lifecycle, you can call
- * <pre>
- * {@code
- *     MixpanelAPI mixpanel = MixpanelAPI.getInstance(...);
- *     mixpanel.getTweaks().defineTweak("Show Advertisement", false);
- * }
- * </pre>
- *
- * Later, you can call
- *
- * <pre>
- * {@code
- *     if (mixpanel.getTweaks().getBoolean("Show Advertisement")) {
- *         showTheAd();
- *     }
- * }
- * </pre>
- *
- * When we defined "Show Advertisement", we gave a default value of false, so to start out, we won't
- * show the ad. However, when you log in to the Mixpanel A/B test editor, you'll be able to see
- * "Show Advertisement", and change it's value to true for some or all of your users.
- *
+ * In general, you won't need to interact with this class directly -
+ * it's used internally to communicate changes in values to the tweaks you define with
+ * {@link MixpanelAPI#stringTweak(String, String)}, {@link MixpanelAPI#booleanTweak(String, boolean)},
+ * {@link MixpanelAPI#doubleTweak(String, double)}, {@link MixpanelAPI#longTweak(String, long)},
+ * and other tweak-related interfaces on MixpanelAPI.
  *
+ * Instances of tweaks aren't available to library user code.
  */
 public class Tweaks {
+    /**
+     * This method is used internally to expose tweaks to the Mixpanel UI,
+     * and will likely not be directly useful to code that imports the Mixpanel library.
+     * The given listener's onTweakDeclared method will be called when a new tweak is declared.
+     */
+    public synchronized void addOnTweakDeclaredListener(OnTweakDeclaredListener listener) {
+        if (null == listener) {
+            throw new NullPointerException("listener cannot be null");
+        }
+        mTweakDeclaredListeners.add(listener);
+    }
+
+    /**
+     * Manually set the value of a tweak. Most users of the library will not need to call this
+     * directly - instead, the library will call set when new values of the tweak are published.
+     */
+    public synchronized void set(String tweakName, Object value) {
+        if (!mTweakValues.containsKey(tweakName)) {
+            Log.w(LOGTAG, "Attempt to set a tweak \"" + tweakName + "\" which has never been defined.");
+            return;
+        }
+
+        final TweakValue container = mTweakValues.get(tweakName);
+        final TweakValue updated = container.updateValue(value);
+        mTweakValues.put(tweakName, updated);
+    }
+
+    /**
+     * Returns the descriptions of all tweaks that currently exist.
+     *
+     * The Mixpanel library uses this method internally to expose tweaks and their types to the UI. Most
+     * users will not need to call this method directly.
+     */
+    public synchronized Map<String, TweakValue> getAllValues() {
+        return new HashMap<String, TweakValue>(mTweakValues);
+    }
 
     @IntDef({
```

Comment:
The Web UI really only knows four types - integral, floating point, boolean, and text. We provide a `byte` type to the user but under the hood it doesn't really matter (and we overflow/truncate anyhow, since we get JSON values)